package.json

```diff
@@ -33,7 +34,9 @@
 		"eslint-plugin-liferayportal": "^1.0.0",
 		"eslint-plugin-no-for-of-loops": "^1.0.0",
 		"eslint-plugin-no-only-tests": "^2.1.0",
-		"eslint-plugin-notice": "0.7.8"
+		"eslint-plugin-notice": "0.7.8",
+		"eslint-plugin-react": "^7.12.4",
+		"eslint-plugin-react-hooks": "^1.5.1"
```

Comment:
hey @wincent I've been thinking about maybe adding these dependencies in `devDependencies` and `peerDependencies` what do you think?

Reply:
My (possibly inaccurate) understanding of the use, based on [the docs](https://docs.npmjs.com/files/package.json) is:

* `dependencies` are for things that your package needs in order to work.
* `devDependencies` are for things that you need in order to develop your package.
* `peerDependencies` are how you declare compatibility with things you expect to be present in the environment your package is installed into.

So, for a developer tool like "eslint-config-liferay", these things are all "dependencies" because without them the package doesn't actually work. But "prettier", for example, is in "devDependencies" because you don't need it to *use* "eslint-config-liferay", only to *develop* it.

My understanding is that if you add "eslint-config-liferay" in "devDependencies", that *doesn't* meant that the things it declares as "dependencies" suddenly become non-dev dependencies for you; they still remain dev-dependencies. It seems that what we have here is a declaration of what this package needs to work, what we need to work on it, and what we expect it to work with.

I looked at [eslint-config-airbnb to see if they follow this pattern or not](https://github.com/airbnb/javascript/blob/master/packages/eslint-config-airbnb/package.json), and the answer is "both" 😂; that is, they have some dependencies in "dependencies", others in "devDependencies" *and* "peerDependencies", and others in just "devDependencies".

"eslint-config-google", on the other hand [only has eslint as a dependency](https://github.com/google/eslint-config-google/blob/master/package.json), and it declares it in "devDependencies"/"peerDependencies".

TL;DR: I think the way we have it is right, but I could easily be mistaken!

Reply:
Yeah, what I thought about it and adding in `peerDependencies` we can let the devs keep those dependencies easily updated without us having to update so often, but the problem of that we have to warn people that they will have to install these dependencies for the package to work as expected (this is a negative point...) another point to add in `peerDependencies` which helps in "deduplicating" the same packages in a project since it will use the package version from the project. I find it a little difficult to happen for this case. 

I think keeping some packages in `devDependencies` is more of the packages we can use locally in the project, as we are doing here. I think it is the use case the other repositories. Let people keep the configuration packages up to date without having to wait for us to update. But I think we can go like this now and see if that's something we'll need in the future...

Reply:
It's definitely a trade-off. Using `peerDependencies` more heavily means:

- Good: Greater control for the user, and the ability to update dependencies pre-emptively without waiting for "eslint-config-liferay" to update them.
- Bad: Less of a batteries-included approach to linting.

I think we should bias towards being a "batteries-included" config, because:

- Ultimately the "people" using this config are actually just us, which means that if we ever need to update the config, nothing will block us.
- We can provide a single, standard linting configuration that we use across all of our projects that is not dependent on whether a particular project has choosen to update its `peerDependencies` or not; you'll just know that if you update to the latest "eslint-config-liferay", you'll always get the current "best practice".
- The ability to update to new versions of dependencies via `peerDependencies` is likely overrated, especially if we use looser ranges (like: `^0.7.8` instead of `0.7.8`).
- Package deduplication is likely a non-issue not just because we have large disks and fast computers, but also partly because we don't expect people to be installing a bunch of competing lint configurations that might specify other versions of our dependencies (this due the package being a "batteries-included" thing), and partly because these dev-time dependencies can't have any negative impact on production-facing things like bundle sizes and so forth.

Reply:
Good points I think it is really better to go without `peerDependencies` for our case. 🙂